src/helpers.cpp

```diff
@@ -105,4 +108,18 @@ int wlcs::helpers::get_argc()
 char const** wlcs::helpers::get_argv()
 {
     return ::argv;
-}
\ No newline at end of file
+}
+
+void wlcs::helpers::set_module_under_test(char const* dso_name)
+{
+    ::dso_name = dso_name;
```

Comment:
Here, or in the calling code, we should check for nullptr. (User input can never be trusted.)

Reply:
I've moved much of this around; we now give actual error messages on invalid input.